code/modules/vehicles/speedbike.dm

```diff
@@ -36,6 +36,118 @@
 	icon_state = "speedbike_red"
 	overlay_state = "cover_red"
 
+// Engineer's repair-bike
+
+/obj/vehicle/space/speedbike/repair
+	desc = "An experimental prototype repair device mounted on a speederbike... what will they think of next."
+	icon_state = "engi_bike"
+	overlay_state = "cover_engi"
+	var/obj/machinery/repair_turret/turret = null
+
+/obj/vehicle/space/speedbike/repair/New()
+	. = ..()
+	turret = new(loc)
+	turret.base = src
+
+/obj/vehicle/space/speedbike/repair/buckle_mob(mob/living/M, force = 0, check_loc = 1)
+	. = ..()
+	riding_datum = new/datum/riding/space/repair
+
+/obj/vehicle/space/speedbike/repair/Move(newloc,move_dir)
+	if(has_buckled_mobs())
+		if(istype(newloc,/turf/open/space))
+			new/turf/open/floor/plating(newloc)
+	. = ..()
+
+
+/obj/machinery/repair_turret
+	name = "mounted repair turret"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "mini_off"
+	density = 0
+	anchored = 1
+	var/atom/base = null
+	var/cooldown = 0
+
+/obj/machinery/porta_turret/New(loc)
+	..()
+	if(!base)
+		base = src
+	update_icon()
+
+/obj/machinery/repair_turret/process()
```

Comment:
I agree its kind of gotten out of hand, for reference does splitting it up into procs just serve as an organizational thing?